x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/action/filter/SecurityActionFilter.java

```diff
@@ -69,17 +69,6 @@ public SecurityActionFilter(AuthenticationService authcService, AuthorizationSer
     public <Request extends ActionRequest, Response extends ActionResponse> void apply(Task task, String action, Request request,
                                                                                        ActionListener<Response> listener,
                                                                                        ActionFilterChain<Request, Response> chain) {
-        /*
-         A functional requirement - when the license of security is disabled (invalid/expires), security will continue
-         to operate normally, except all read operations will be blocked.
-         */
-        if (licenseState.checkFeature(Feature.SECURITY_STATS_AND_HEALTH) == false && LICENSE_EXPIRATION_ACTION_MATCHER.test(action)) {
-            logger.error("blocking [{}] operation due to expired license. Cluster health, cluster stats and indices stats \n" +
-                    "operations are blocked on license expiration. All data operations (read and write) continue to work. \n" +
-                    "If you have a new license, please update it. Otherwise, please reach out to your support contact.", action);
-            throw LicenseUtils.newComplianceException(XPackField.SECURITY);
-        }
```

Comment:
My understanding on the intention of this piece of code is that we do not encourage users to run with expired commerical licenses, e.g. PLATINUM, forever. An expired PLATINUM license is still allowed to access some platinum features, e.g. DLS/FLS, for security reasons. If an expired license can access all BASIC featues, plus a fair bit more, this would put our licensing in an awkward situation. Therefore, we disable stats and health APIs to prompt users to either downgrade to a BASIC or renew the subscription.

In fact, many of the BASIC features have similar behaviour as the stats and health API, e.g. `ROLLUP(OperationMode.MISSING, true)`. The feature is available in BASIC, but also requires the license to be active.